R/ui.R

```diff
@@ -0,0 +1,114 @@
+#' The application User-Interface
+#'
+#' @param request Internal parameter for `{shiny}`.
+#' @noRd
+#' @import shiny
+#' @import bslib
+#' @import bsicons
+#' @importFrom tibble as_tibble
+#' @importFrom DT DTOutput
+#' @importFrom plotly plotlyOutput
+
+
+app_ui <- function(request) {
+  page_sidebar(
+    theme = bs_theme(version = 5, bootswatch = "lumen"),
+    title = "Code Usage Explorer",
+    sidebar = sidebar(
+      card(
+        card_header("Select dataset"),
+        radioButtons("dataset", NULL,
+          choices = c(
+            "SNOMED-CT" = "snomed",
+            "ICD-10" = "icd10",
+            "OPCS" = "opcs"
+          )
+        )
+      ),
+      card(
+        card_header("Select codes"),
+        selectizeInput(
+          "code_search",
+          "Specific code",
+          choices = NULL,
+          multiple = TRUE,
+          options = list(maxOptions = 15)
+        ),
+        textInput(
+          "description_search",
+          "Description",
+          ""
+        )
+      ),
+      card(
+        card_header(
+          tooltip(
+            span(
+              "Load from OpenCodelist",
+              bs_icon("info-circle")
+            ),
+            "Enter <codelist_id>/<version_id>, for example: 'nhsd-primary-care-domain-refsets/bp_cod/20200812'",
+            options = list(
+              customClass = "left-align-tooltip"
+            )
+          )
+        ),
+        textInput(
+          "codelist_slug",
+          "Codelist ID / Version Tag",
+          placeholder = "nhsd-primary-care-domain-refsets/bp_cod/20200812",
```

Comment:
I still wonder if copying in the whole URL is not simplest from the user perspective (just one copy and paste), as doesn't require manually deleting the "https://" etc.